src/FileIO.chpl

```diff
@@ -359,15 +353,12 @@ module FileIO {
       var (strictFlag, ndsetsStr, nfilesStr, allowErrorsFlag, calcStringOffsetsFlag, arraysStr) = payload.splitMsgToTuple(6);
       var (jsondsets, jsonfiles) = arraysStr.splitMsgToTuple(" | ",2);
 
-      if (!checkCast(nfilesStr, int)) {
-        var errMsg = "Number of files:`%s` could not be cast to an integer".format(nfilesStr);
-        return new MsgTuple(errMsg, MsgType.ERROR);
-      }
-      var nfiles = nfilesStr:int; // Error checked above
+      var msgArgs = parseMessageArgs(payload, 7);
+      var nfiles = msgArgs.get("filename_size").getIntValue();
       var filelist: [0..#nfiles] string;
       
       try {
-        filelist = jsonToPdArray(jsonfiles, nfiles);
+        filelist = msgArgs.get("filenames").getList(nfiles);
```

Comment:
we are unpacking 7 things into `msgArgs` but it looks like we only access `filename_size` and `filenames`. I think the others are being grabbed from the `splitMsgToTuple` lines?

Reply:
Actually in this message, most of the parameters are not used until you go into the HDF5/Parquet specific processing.